drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1350,7 +1350,7 @@ static uint32_t reverse(uint32_t val)
 	int i;
 
 	for (i = 0; i < 32; i++) {
-		if (val & (1 << i)) {
+		if (val & (1UL << i)) {
```

Comment:
Could be done using existing utils `BIT` define.

```suggestion
		if (val & BIT(n)) {
```

Reply:
Updated in new version.